Automattic/jetpack
# Boost: Explicitly hide admin-bar when generating Critical CSS

Fixes #20549

When generating Critical CSS on Atomic sites, the admin-bar was getting included in the resultant CSS. When viewing the site with `jb-generate-critical-css` in the URL, you could see it, too.

The problem is that the admin bar is forced "on" when viewing Atomic sites via the proxy, to show the proxied status.

This PR addresses (part of) the problem by adding a hook to explicitly turn off the admin-bar. This should be harmless on non-atomic sites, and may catch other edge-cases in the wild.

Note: When generating via the proxy, the "proxied" badge will still be included - but that is only relevant to a12s.

#### Changes proposed in this Pull Request:
* Use the `show_admin_bar` hook with a very low priority to explicitly turn off the admin bar during CSS generation.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Generate Critical CSS on your testing site.
* Visit the `?jb-generate-critical-css` URLs and ensure nothing breaks, and no admin bars are shown. Especially on Atomic sites.
